### PR TITLE
feat(snowflake): implement  `random` scalar

### PR DESCRIPTION
**Pull request by mik-laj**
Hello,

``ibis.random()`` will be expressed as the following SQL:
````sql
SELECT uniform(CAST(0.0 AS FLOAT), CAST(1.0 AS FLOAT), random()) AS "RandomScalar()"
````
Best regards,

### CODE REVIEW

**Review by cpcloud — APPROVED** [review]
LGTM, merging on green!

Thank you!

File: ibis/backends/tests/test_numeric.py [hunk 1]
```diff
@@ -527,20 +527,14 @@ def test_sa_default_numeric_precision_and_scale(
 
 @pytest.mark.notimpl(["dask", "datafusion", "impala", "pandas", "sqlite", "polars"])
 @pytest.mark.notyet(
-    ["clickhouse", "snowflake"],
-    reason="backend doesn't implement a [0.0, 1.0) random function",
+    ["clickhouse"],
+    reason="backend doesn't implement a [0.0, 1.0) or [0.0, 1.0] RANDOM() function",
```
Comment (mik-laj) [inline ibis/backends/tests/test_numeric.py:531]: Should we also update docstring of `random` function?
https://github.com/ibis-project/ibis/blob/409f49709d90ffb1af480cfa4dd8ea62597cba32/ibis/expr/api.py#L602

Comment (cpcloud) [reply ibis/backends/tests/test_numeric.py:531]: Good idea! We can do that in a follow-up.

**Comment by cpcloud** [comment]
Thanks @mik-laj!

**Pull request merged**